store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -570,6 +573,15 @@ private Element doResponse(Element request, AuthToken at, ZimbraSoapContext zsc,
         if (acct.isIsMobileGatewayProxyAccount()) {
             response.addAttribute(AccountConstants.A_ZMG_PROXY, true);
         }
+
+        ToXML.encodeAttr(response, Provisioning.A_zimbraTwoFactorAuthMethodAllowed, acct.getMultiAttr(Provisioning.A_zimbraTwoFactorAuthMethodAllowed));
+        ToXML.encodeAttr(response, Provisioning.A_zimbraTwoFactorAuthMethodEnabled, acct.getMultiAttr(Provisioning.A_zimbraTwoFactorAuthMethodEnabled));
+        ToXML.encodeAttr(response, Provisioning.A_zimbraPrefPrimaryTwoFactorAuthMethod, acct.getAttr(Provisioning.A_zimbraPrefPrimaryTwoFactorAuthMethod));
+        String recoveryAddrs = acct.getAttr(Provisioning.A_zimbraPrefPasswordRecoveryAddress);
+        if (!StringUtil.isNullOrEmpty(recoveryAddrs)) {
+            String prefPwdRecoveryAddrs = Stream.of(recoveryAddrs.split(",")).map(add -> StringUtil.maskEmail(add)).collect(Collectors.joining(","));
```

Comment:
I have three questions.
1. Why don't we use auto-generated getters like `getTwoFactorAuthMethodAllowed`?
The getters return a default value considering domain/cos-inherited when no value has been set on the account.
https://github.com/Zimbra/zm-mailbox/pull/1519/files
```
    @ZAttr(id=4102)
    public String[] getTwoFactorAuthMethodAllowed() {
        String[] value = getMultiAttr(Provisioning.A_zimbraTwoFactorAuthMethodAllowed, true, true); return value.length > 0 ? value : new String[] {"app"};
    }
```

2. What does `recoveryAddrs.split(",")` mean?
Is there any case that zimbraPrefPasswordRecoveryAddress has comma-separated multiple email addresses?
I thought it could have only a single email address.

3. What does `Pwd` mean in a variable name `prefPwdRecoveryAddrs`?

Reply:
1 - We can also use `getTwoFactorAuthMethodAllowed` as both have same implementation. 
acct.getMultiAttr(Provisioning.A_zimbraTwoFactorAuthMethodAllowed)) will also return the default value when no value has been set on the account. 
2 - Yes.. zimbraPrefPasswordRecoveryAddress is type of cs_emailp and you can see the description -
`cs_emailp...comma-separated valid email addresses . each address must have a "@" 
                and personal part is optional.`
3 - Pwd is referring password here, wasn't wanted to have long name. You can suggest a good name if doesn't sound good.

Reply:
Thank you for your answers. I understood.